### ISSUE TITLE
Add a branch without an upper PHP version limit

### ISSUE THREAD

**Issue by oliverklee** [MyIntervals/emogrifier#1210]
Also automatically rebase it from `main`.

**Comment by JakeQZ** [comment]
As I understand, this would be a branch available via Composer like the `-dev` branch, but would not have any 'releases'.  Or could it have automatic mirrored releases, with some kind of suffix (like there is `-alpha` and `-beta` - though I'm not sure what suffix we should choose for these, or if there is a precedent set by other projects)?  People would then be able to use it via Composer in [situations where the upper PHP version limit is restrictive in ways we haven't foreseen](https://github.com/MyIntervals/emogrifier/pull/1208#issuecomment-1654769689).

**Comment by oliverklee** [comment]
> As I understand, this would be a branch available via Composer like the `-dev` branch, but would not have any 'releases'. 

Exactly. And we can regularly rebase it on the `main` branch (or automate that via a GitHub action).

**Comment by oliverklee** [comment]
AFAIK, there is no naming convention for this.

**Comment by mvorisek** [comment]
> @mvorisek, thanks for sharing your experience; useful feedback and good to know. Feel free to further comment on https://github.com/MyIntervals/emogrifier/issues/1210 there.

I maintain several large php projects and the simplest and most powefull solution is to not drop older PHP versions support too early. The PHP changes needed between PHP 7.4 and PHP 8.0 are minimal. The only limitation is how early a new PHP features can be uses, but they are mostly syntactical sugars...

The most important thing is to test everyting using CI - here is an example https://github.com/atk4/core/blob/9a3ff6928/.github/workflows/test-unit.yml#L80 which tests also if the lowest deps pass all tests.
